### PR TITLE
chore(utxo-lib): uprgade tiny-secp256k1

### LINKED ISSUE #7612: Dependencies: tiny-secp256k1
Update `tiny-secp256k1`

Issues:
- not found changes in v2
- failing tests
- enable `WebAssembly` in `webpack`
Screenshot_2023-02-10_at_16 36 00

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/7612

### CODE REVIEW

**Comment by komret** [comment]
Keep going! 💪

File: packages/suite-build/configs/base.webpack.config.ts [hunk 1]
```diff
@@ -197,6 +197,8 @@ const config: webpack.Configuration = {
               ]
             : []),
     ],
+    // For tiny-secp256k1
+    experiments: { asyncWebAssembly: true },
```
Comment (szymonlesisz) [inline packages/suite-build/configs/base.webpack.config.ts:201]: is it required by tiny-secp256k1 or optional? if required then this should be added to all the projects which are using utxo-lib (connect, connect-iframe, connect-popup, connect-explorer)

i know that we build some other wasm files (cardano?), will this have effect on them as well?

File: packages/utxo-lib/src/bip32.ts [hunk 2]
```diff
@@ -131,13 +132,15 @@ class BIP32 implements BIP32Interface {
...
+        if (this.__Q === undefined && this.__D !== undefined) {
+            this.__Q = uin8ArrayToBuffer(ecc.pointFromScalar(this.__D, true)) ?? undefined;
+        }
 
         return this.__Q!;
     }
 
     get privateKey(): Buffer | undefined {
-        return this.__D;
+        return uin8ArrayToBuffer(this.__D) ?? undefined;
```
Comment (szymonlesisz) [inline packages/utxo-lib/src/bip32.ts:143]: is this needed? `this.__D` is defined as Buffer and should be a Buffer.

this comment applies for all the usage of uin8ArrayToBuffer in this file

**Review by szymonlesisz — COMMENTED** [review]
as we introduce new external file (wasm) this change should be carefully tested in multiple enviroments (for example by signing testnet transaction)

on desktop:
suite-web, suite-desktop, connect-popup, connect-webextension, connect-explorer

on mobile:
suite-mobile app, suite-web in browser, connect-popup in browser

File: packages/utxo-lib/src/uin8ArrayToBuffer.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    // eslint-disable-next-line no-nested-ternary
+    (array === null
+        ? null
+        : array === undefined
+          ? undefined
+          : Buffer.from(array)) as T extends Uint8Array
+        ? Buffer
+        : T extends null
+          ? null
+          : undefined;
```
Comment (mroz22) [inline packages/utxo-lib/src/uin8ArrayToBuffer.ts:13]: note te me - to be discussed

**Pull request closed without merging**